lib/src/source/hosted.dart

```diff
@@ -212,8 +252,11 @@ class HostedSource extends Source {
       url = validateAndNormalizeHostedUrl(u);
     }
 
-    return Pair<String, Uri>(name, url);
+    return _HostedDescription(name, url);
   }
+
+  static const LanguageVersion _minVersionForShorterHostedSyntax =
```

Comment:
Adding a few comments here might help future generations maintain this code :D 

```suggestion
  /// Minimum language version at which short hosted syntax is supported.
  ///
  /// This allows `hosted` dependencies to be expressed as:
  /// ```yaml
  /// dependencies:
  ///   foo:
  ///     hosted: https://some-pub.com/path
  ///     version: ^1.0.0
  /// ```
  static const LanguageVersion _minVersionForShortHostedSyntax =
```